spec/models/user_spec.rb

```diff
@@ -74,11 +74,31 @@ def user_error_message(*keys)
         )
 
         expect(SidebarSectionLink.where(linkable_type: 'Tag', user_id: user.id).pluck(:linkable_id)).to contain_exactly(
-          tag.id,
-          hidden_tag.id
+          tag.id
         )
       end
 
+      it "Should only receive new categories they didn't previously have access to" do
+        user = Fabricate(:user)
+        SidebarSectionLink.where(user: user).delete_all # User has customized their sidebar categories
```

Comment:
I think we should leave the existing sidebar section links in, what we want to test is that the secured categories are added and removed on top of the existing sidebar section links. If we delete everything here, we're unable to fully test that.